src/tools/tidy/src/ui_tests.rs

```diff
@@ -73,6 +79,56 @@ pub fn check(root_path: &Path, tidy_ctx: TidyCtx) {
             ));
         }
     }
+
+    // The list of subdirectories in ui tests.
+    // Compare previous subdirectory with current subdirectory
+    // to sync with `tests/ui/README.md`.
+    // See <https://github.com/rust-lang/rust/issues/150399>
+    let mut prev_line = "";
+    let mut is_sorted = true;
+    let prev_subdirs: BTreeSet<_> = include_str!("ui_suites.txt")
+        .strip_prefix(UI_SUITES_TXT_HEADER)
+        .unwrap()
+        .lines()
+        .inspect(|&line| {
+            if prev_line > line {
+                is_sorted = false;
+            }
+            prev_line = line;
+        })
+        .map(|line| line.to_string())
+        .collect();
```

Comment:
The logic to check if it is sorted is duplicated in issues.txt. Do I need to separate it into a function?

Reply:
Yes, I refactored it.